### ISSUE TITLE
Kill-aura entities still visible

### ISSUE THREAD

**Issue by elstur** [MeGysssTaa/ReflexIssueTracker#188]
Spigot version 1.8.8 (latest)
Reflex version 10.2-3U (latest)

Unless I disable kill-aura checks completely there is no setting or way to make the npc's behind you invisible.
Even setting the visible timer to -1 their nametags are still visibly seen floating around behind. I have tried making changes to the name-tag settings to adjust this. I have no mods that alter name-tags.

I don't have the ability to record at this time to show, but they are floating behind you.

**Comment by MeGysssTaa** [comment]
Try disabling "realistic_playerdata" in killaura settings. If that won't help, than you're running an incompatible Nametags plugin.

**Comment by elstur** [comment]
As I said above, I have no mods that alter nametags and toggling those settings didn't work. I'll just keep kill-aura disabled.

As an interesting side note, Only 1.8 clients see the tags. Any other client does not see them.

**Comment by MeGysssTaa** [comment]
You are running **PLUGINS** that change players' nametags on your **server**, I haven't meant client mods.

**Comment by elstur** [comment]
Please don't be rude. As I said 2 other times I am not running any mods (PLUGINS) on my **server** that alter/change/impact nametags. Why on earth would you assume I am talking about client-sided mods on a server side issue?

As an added test, I ran a clean install of the latest version of spigot 1.8, set NPC visibility to -1, disabled "realistic_playerdata". Reflex was the only PLUGIN installed on this server. I can still see their nametags when in combat floating behind me.

**Comment by MeGysssTaa** [comment]
Sorry if I seemed rude, I just highlighted "plugins" because you were actually mentioning "mods", which is a different thing. Nametags of invisible entities are not displayed in vanilla Minecraft itself. The reasons you can see them can be only one of:

- You're using some client/server mods that affect nametags.
- You're using some server plugins that affect nametags.
- You're using a modified server core which has nametags system modified.
- Your server world (map) forces all players joining a team, which's options allow teammates to see each other invisible.

I cannot help you unless you do a proper bug report (with /reflex debug and videos), sadly.

**Comment by elstur** [comment]
I will do a proper report when I get a chance.

As a side note, I have a 1.11 survival server which also has no nametag plugins installed. Same settings and in 1.8 clients can see the floating nametags but no other clients can. Is it possibly an issue with 1.8 clients?

-----

Now, I am planning on building a mod on my kitpvp server that will modify nametags to show that players are in-combat (have not done this yet). Is there an API method I can use to prevent this from making reflex kill-aura detection visible?

**Comment by MeGysssTaa** [comment]
> As a side note, I have a 1.11 survival server which also has no nametag plugins installed. Same settings and in 1.8 clients can see the floating nametags but no other clients can. Is it possibly an issue with 1.8 clients?
1.11 server that allows joining of 1.8 clients?

> Is there an API method I can use to prevent this from making reflex kill-aura detection visible?
Just make your plugin ignoring invisible entities and only create nametags for visible ones.

**Comment by elstur** [comment]
Yes, it uses protocolsupport to allow 1.8-1.12 clients to join. I want my entire server to have 1.8-1.12 but don't want to hold back my 1.12 client players. Everything operates fine. Kill-aura detection isn't a need on this survival server, but of course is never a bad thing to have.
